### PR TITLE
Migrated from React.PropTypes to prop-types package, to support React 15.5.x

### PR DESCRIPTION
**Pull request by SuhairZain**
#### What does this PR do?
From React 15.5.x, PropTypes has been moved to a separate package. Did the migration using JSCodeShift and a helper script.

#### What testing has been done on this PR?
Ran pre-commit tests

#### What are the relevant issues?
https://github.com/grommet/grommet/issues/1303

#### Do the grommet docs need to be updated?
No

#### Should this PR be mentioned in the release notes?
No

#### Is this change backwards compatible or is it a breaking change?
Backwards compatible

### CODE REVIEW

File: src/js/components/Anchor.js [hunk 1]
```diff
@@ -1,8 +1,9 @@
 // (C) Copyright 2014-2016 Hewlett Packard Enterprise Development LP
 
+import PropTypes from 'prop-types';
 import React, { Children, Component } from 'react';
 import classnames from 'classnames';
-import { schema, PropTypes } from 'react-desc';
```
Comment (alansouzati) [inline src/js/components/Anchor.js]: I believe we should keep the PropTypes coming from react-desc

File: src/js/components/Accordion.js [hunk 2]
```diff
@@ -1,6 +1,7 @@
 // (C) Copyright 2014-2016 Hewlett Packard Enterprise Development LP
 
-import React, { Component, PropTypes } from 'react';
+import PropTypes from 'prop-types';
```
Comment (alansouzati) [inline src/js/components/Accordion.js]: Can we add React first and PropTypes after ?

**Comment by SuhairZain** [comment]
I noticed that too. I thought that since the migration was done by the [React Community script](https://github.com/reactjs/react-codemod/blob/master/transforms/React-PropTypes-to-prop-types.js), `prop-types` were being migrated from `react-desc` as well.

If it isn't, I think we should raise an issue in the JsCodeShift repo. Could be a bug.

File: src/js/components/Accordion.js (hunk 2 above)
Comment (SuhairZain) [inline src/js/components/Accordion.js]: Yes. It'd be quite easy given that React import is done in a consistent manner in Grommet. 

I'm travelling right now. Will spin up a script in some time and push.

Comment (alansouzati) [inline src/js/components/Accordion.js]: no rush :)

**Comment by alansouzati** [comment]
I just updated react-desc to support react 15.5:

https://github.com/grommet/react-desc/commit/bf69c63605bbbdf7d1a261535a54e93758d0263a

**Comment by alansouzati** [comment]
Also all the icons inside `base` folder are automatically generated. you can ignore them in your refactor. I've updated grommet-icon-loader to use the latest React also.

https://github.com/grommet/grommet-icon-loader/commit/efd8279c2094b1bddaaa9105d9652c94e1f6d38e

**Comment by SuhairZain** [comment]
I have re-ordered the imports using a Regex and avoided the icons/base folder in the refactoring. Only 93 files are changed in this commit. 😄

**Comment by alansouzati** [comment]
hey @SuhairZain is still see 584 files in this commit. can you make sure base will not be there?

**Comment by SuhairZain** [comment]
Have made another PR with a clean branch touching only just the 96 component files and 2 other files(package.json and yarn.lock). Kindly check if it's okay.

**Pull request closed without merging**